src/Compilers/CSharp/Portable/Operations/CSharpOperationNodes.cs

```diff
@@ -10,6 +11,23 @@
 
 namespace Microsoft.CodeAnalysis.Operations
 {
+
+    internal sealed class CSharpLazyNoneOperation : LazyNoneOperation
+    {
+        private readonly Func<BoundNode, ImmutableArray<IOperation>> _getChildren;
```

Comment:
This is still capturing. We should instead hold onto the `CSharpOperationFactory` here, and call `GetIOperationChildren` off that instance below.